data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -194,47 +212,52 @@ private void resetOffsets() {
     }
 
     void processAcknowledgedOffsets() {
-        synchronized(acknowledgedOffsets) {
```

Comment:
why is this synchronized block removed ?

Reply:
The calling function has `synchronized(this)`

Reply:
This is from Krishna's CR https://github.com/opensearch-project/data-prepper/pull/3200/files
synchronization is moved to calling function.